sdk/storage/Azure.Storage.Common.DataMovement/src/DirectoryScanner.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace Azure.Storage
+{
+    internal class DirectoryScanner
+    {
+        private string _path;
+
+        public DirectoryScanner(string path)
+        {
+            _path = path;
+        }
+
+        // Constuctor for mocking
+        public DirectoryScanner() { }
+
+        public IEnumerable<string> EnumerateDirectories()
```

Comment:
Why do we call this EnumerateX while we call it Scan elsewhere?

Would it be better to return some model from here which has both name and type (file vs directory) and collapse these methods?

Reply:
This was just making an abstraction of the built in methods so these could be mocked. Now that I'm thinking about it, these helper abstractions don't necessitate a whole new class; I can move these into private methods inside the FileSystemScanner.